### PR TITLE
refactor: optimized CLI usage and adjust default builds

### PR DESCRIPTION
**Pull request by venjiang**
# Description

Optimized CLI usage and adjust default builds.

## Impact

- `yomo build` command removes the `target` flag and builds `wasm` by default.
eg. `yomo build --target wasm app.go` => `yomo build app.go`
- `yomo run` command runs only the `wasm` target.
- `yomo dev` command runs only the `wasm` target.
- Remove 'exec' serverless.

## Example
For details, please visit `example/10-wacli`

### CODE REVIEW

File: cli/dev.go [hunk 1]
```diff
@@ -28,17 +28,16 @@ var devViper *viper.Viper
 
 // devCmd represents the dev command
 var devCmd = &cobra.Command{
-	Use:                "dev",
+	Use:                "dev [flags] sfn.wasm",
 	Short:              "Dev a YoMo Stream Function",
 	Long:               "Dev a YoMo Stream Function with mocking yomo-source data from YCloud.",
```
Comment (fanweixiao) [inline cli/dev.go:33]:
```suggestion
	Long:               "Test a YoMo Stream Function with public zipper and mocking data",
```

File: cli/dev.go [hunk 2]
```diff
@@ -28,17 +28,16 @@ var devViper *viper.Viper
 
 // devCmd represents the dev command
 var devCmd = &cobra.Command{
-	Use:                "dev",
+	Use:                "dev [flags] sfn.wasm",
 	Short:              "Dev a YoMo Stream Function",
```
Comment (fanweixiao) [inline cli/dev.go:32]:
```suggestion
	Short:              "Test a YoMo Stream Function",
```

**Review by fanweixiao — COMMENTED** [review]
👍🏼

**Review by fanweixiao — APPROVED** [review]

**Review by woorui — APPROVED** [review]

File: cli/build.go [hunk 3]
```diff
@@ -73,12 +59,8 @@ var buildCmd = &cobra.Command{
 func init() {
 	rootCmd.AddCommand(buildCmd)
 
-	buildCmd.Flags().StringVarP(&opts.Filename, "file-name", "f", "app.go", "Stream function file (default is app.go)")
-	// buildCmd.Flags().StringVarP(&url, "url", "u", "localhost:9000", "YoMo-Zipper endpoint addr")
-	// buildCmd.Flags().StringVarP(&opts.Name, "name", "n", "", "yomo stream function app name (required). It should match the specific service name in YoMo-Zipper config (workflow.yaml)")
 	buildCmd.Flags().StringVarP(&opts.ModFile, "modfile", "m", "", "custom go.mod")
-	// buildCmd.Flags().StringVarP(&opts.Credential, "credential", "d", "", "client credential payload, eg: `token:dBbBiRE7`")
-	buildCmd.Flags().StringVarP(&opts.Target, "target", "t", "", "Build target: native or wasm")
+	buildCmd.Flags().StringVarP(&opts.Builder, "builder", "b", "tinygo", "Builder: use native gojs or tinygo")
```
Comment (fanweixiao) [inline cli/build.go:63]: Remove this

**Review by wujunzhuo — APPROVED** [review]

**Pull request merged**